error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/BigDecimalRules.java

```diff
@@ -66,4 +66,114 @@ BigDecimal after(double value) {
       return BigDecimal.valueOf(value);
     }
   }
+
+  /** Prefer using {@link BigDecimal#signum()} over comparing to {@link BigDecimal#ZERO}. */
+  static final class BigDecimalCompareToZero {
+    @BeforeTemplate
+    boolean before(BigDecimal value) {
+      return Refaster.anyOf(
+          value.compareTo(BigDecimal.ZERO) == 0, BigDecimal.ZERO.compareTo(value) == 0);
+    }
+
+    @AfterTemplate
+    boolean after(BigDecimal value) {
+      return value.signum() == 0;
+    }
+  }
+
+  /** Prefer using {@link BigDecimal#signum()} over comparing to {@link BigDecimal#ZERO}. */
+  static final class BigDecimalCompareToNonZero {
+    @BeforeTemplate
+    boolean before(BigDecimal value) {
+      return Refaster.anyOf(
+          value.compareTo(BigDecimal.ZERO) != 0, BigDecimal.ZERO.compareTo(value) != 0);
+    }
+
+    @AfterTemplate
+    boolean after(BigDecimal value) {
+      return value.signum() != 0;
+    }
+  }
+
+  /** Prefer using {@link BigDecimal#signum()} over comparing to {@link BigDecimal#ZERO}. */
+  static final class BigDecimalCompareToZeroPositive {
+    @BeforeTemplate
+    boolean before(BigDecimal value) {
+      return Refaster.anyOf(
+          value.compareTo(BigDecimal.ZERO) > 0, BigDecimal.ZERO.compareTo(value) < 0);
+    }
+
+    @AfterTemplate
+    boolean after(BigDecimal value) {
+      return value.signum() == 1;
```

Comment:
The `BigDecimalPositiveSignum` has the same after template. We can combine them in the `Refaster#anyOf` 😄.

Reply:
In that case we also need to update the Javadoc :).